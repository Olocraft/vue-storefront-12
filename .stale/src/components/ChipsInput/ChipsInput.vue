<template>
  <div
    class="rounded-full group pl-0.5 pr-1 py-0.5 text-gray-900 bg-white border-2
    border-primary-500 font-body font-normal inline-flex items-center
    align-center w-max cursor-pointer transition duration-300 ease outline-violet"
    :class="{'!pr-3 bg-gray-100 border-gray-200 opacity-50 cursor-not-allowed': disabled, '!pr-0.5': !value}"
  >
    <div :class="chipsIconSize">
      <slot name="icon" />
    </div>
    <span v-if="value" class="ml-1" :class="chipsFontSize">{{ value }}</span>
    <slot name="close">
      <button
        v-if="!disabled"
        v-focus
        class="bg-transparent"
        :class="[chipsButtonSpacing, hideButton]"
        @click="$emit('chips-close')"
      >
        <!-- eslint-disable max-len -->
        <svg
          class="transition duration-300 fill-gray-500 group-hover:fill-primary-600 group-active:fill-primary-700 ease"
          :class="[chipsSize]"
          :aria-label="`Close ${value} chip`"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.1964 4.74332C12.456 4.48375 12.456 4.0629 12.1964 3.80332C11.9368 3.54375 11.516 3.54375 11.2564 3.80332L7.99975 7.05999L4.74308 3.80332C4.4835 3.54375 4.06265 3.54375 3.80308 3.80332C3.5435 4.0629 3.5435 4.48375 3.80308 4.74332L7.05975 7.99999L3.80308 11.2567C3.5435 11.5162 3.5435 11.9371 3.80308 12.1967C4.06265 12.4562 4.4835 12.4562 4.74308 12.1967L7.99975 8.93999L11.2564 12.1967C11.516 12.4562 11.9368 12.4562 12.1964 12.1967C12.456 11.9371 12.456 11.5162 12.1964 11.2567L8.93975 7.99999L12.1964 4.74332Z"
          />
          <path
            d="M12.1964 4.74332C12.456 4.48375 12.456 4.0629 12.1964 3.80332C11.9368 3.54375 11.516 3.54375 11.2564 3.80332L7.99975 7.05999L4.74308 3.80332C4.4835 3.54375 4.06265 3.54375 3.80308 3.80332C3.5435 4.0629 3.5435 4.48375 3.80308 4.74332L7.05975 7.99999L3.80308 11.2567C3.5435 11.5162 3.5435 11.9371 3.80308 12.1967C4.06265 12.4562 4.4835 12.4562 4.74308 12.1967L7.99975 8.93999L11.2564 12.1967C11.516 12.4562 11.9368 12.4562 12.1964 12.1967C12.456 11.9371 12.456 11.5162 12.1964 11.2567L8.93975 7.99999L12.1964 4.74332Z"
          />
        </svg>
        <!-- eslint-enable max-len -->
      </button>
    </slot>
  </div>
</template>

<script>
import { computed } from '@vue/composition-api';
import focus from '../../utils/focus-directive';

export default {
  name: 'ChipsInput',
  directives: {
    focus,
  },
  props: {
    value: {
      type: String,
      default: '',
    },
    size: {
      type: String,
      default: 'medium',
      validator(value) {
        return [
          'small',
          'medium',
          'large',
        ].includes(value);
      },
    },
    disabled: {
      type: Boolean,
      default: false,
    },
  },
  setup(props) {
    const chipsSize = computed(() => {
      switch (props.size) {
        case 'small':
          return 'w-4 h-4';
        case 'large':
          return 'w-5 h-5';
        default:
          return 'w-4 h-4';
      }
    });
    const chipsIconSize = computed(() => {
      switch (props.size) {
        case 'small':
          return 'w-5 h-5';
        case 'large':
          return 'w-8 h-8';
        default:
          return 'w-6 h-6';
      }
    });
    const chipsFontSize = computed(() => {
      switch (props.size) {
        case 'small':
          return 'text-sm';
        case 'large':
          return 'text-base';
        default:
          return 'text-sm';
      }
    });
    const chipsButtonSpacing = computed(() => {
      switch (props.size) {
        case 'small':
          return 'ml-2 mr-1';
        case 'large':
          return 'ml-3 mr-1.5';
        default:
          return 'ml-2.5 mr-1';
      }
    });
    const hideButton = computed(() => (!props.value ? 'hidden' : 'block'));
    return {
      chipsSize,
      chipsIconSize,
      chipsFontSize,
      chipsButtonSpacing,
      hideButton,
    };
  },
};
</script>
