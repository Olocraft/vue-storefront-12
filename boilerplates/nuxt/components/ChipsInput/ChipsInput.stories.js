import ChipsInput from './ChipsInput.vue';

export default {
  title: 'SFUI Components/ChipsInput',
  component: ChipsInput,
  parameters: {
    docs: {
      description: {
        component:
          'The chips component. [link to component docs on the page]'
      }
    }
  },
  argTypes: {
    value: {
      control: 'text',
      table: {
        category: 'Props'
      },
      defaultValue: 'Label',
      description: 'Chips value'
    },
    size: {
      control: 'select',
      defaultValue: 'medium',
      options: ['small', 'medium', 'large'],
      table: {
        category: 'Props'
      },
      description:
        'Size of chips.'
    },
    disabled: {
      control: 'boolean',
      defaultValue: false,
      table: {
        category: 'Props'
      },
      description: 'Chips disabled property'
    }
  }
};

const Template = (args, { argTypes }) => ({
  components: { ChipsInput },
  props: Object.keys(argTypes),
  template: `
    <ChipsInput :value="value" :size="size" :disabled="disabled">
    <template #icon>
      <svg class="w-full h-full fill-gray-500" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
        <path d="M22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12Z"/>
      </svg>
    </template>
    </ChipsInput>`
});

export const Base = Template.bind({});
Base.args = {
  size: 'medium'
};
