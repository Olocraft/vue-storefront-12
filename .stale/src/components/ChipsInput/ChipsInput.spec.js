import { shallowMount } from '@vue/test-utils';
import ChipsInput from './ChipsInput.vue';

let wrapper;

describe('ChipsInput.vue', () => {
  beforeEach(() => {
    wrapper = shallowMount(ChipsInput);
  });
  it('render component correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
