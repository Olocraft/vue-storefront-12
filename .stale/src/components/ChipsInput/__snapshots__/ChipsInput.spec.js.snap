// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipsInput.vue render component correctly 1`] = `
<div class="rounded-full group pl-0.5 pr-1 py-0.5 text-gray-900 bg-white border-2
  border-primary-500 font-body font-normal inline-flex items-center
  align-center w-max cursor-pointer transition duration-300 ease outline-violet !pr-0.5">
  <div class="w-6 h-6"></div>
  <!----> <button class="bg-transparent ml-2.5 mr-1 hidden"><svg aria-label="Close  chip" viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg" class="transition duration-300 fill-gray-500 group-hover:fill-primary-600 group-active:fill-primary-700 ease w-4 h-4">
      <path d="M12.1964 4.74332C12.456 4.48375 12.456 4.0629 12.1964 3.80332C11.9368 3.54375 11.516 3.54375 11.2564 3.80332L7.99975 7.05999L4.74308 3.80332C4.4835 3.54375 4.06265 3.54375 3.80308 3.80332C3.5435 4.0629 3.5435 4.48375 3.80308 4.74332L7.05975 7.99999L3.80308 11.2567C3.5435 11.5162 3.5435 11.9371 3.80308 12.1967C4.06265 12.4562 4.4835 12.4562 4.74308 12.1967L7.99975 8.93999L11.2564 12.1967C11.516 12.4562 11.9368 12.4562 12.1964 12.1967C12.456 11.9371 12.456 11.5162 12.1964 11.2567L8.93975 7.99999L12.1964 4.74332Z"></path>
      <path d="M12.1964 4.74332C12.456 4.48375 12.456 4.0629 12.1964 3.80332C11.9368 3.54375 11.516 3.54375 11.2564 3.80332L7.99975 7.05999L4.74308 3.80332C4.4835 3.54375 4.06265 3.54375 3.80308 3.80332C3.5435 4.0629 3.5435 4.48375 3.80308 4.74332L7.05975 7.99999L3.80308 11.2567C3.5435 11.5162 3.5435 11.9371 3.80308 12.1967C4.06265 12.4562 4.4835 12.4562 4.74308 12.1967L7.99975 8.93999L11.2564 12.1967C11.516 12.4562 11.9368 12.4562 12.1964 12.1967C12.456 11.9371 12.456 11.5162 12.1964 11.2567L8.93975 7.99999L12.1964 4.74332Z"></path>
    </svg></button>
</div>
`;
